src/read_config.c

```diff
@@ -64,36 +67,40 @@ int my_read_int(char *inifile, char *name, int *value)
 
 static int my_read_ulong(char *inifile, char *name, unsigned long long int *value)
 {
-  FILE *in;
-  char buf[100];
-  int found=0;
-
-  in=fopen(inifile,"r");
-  if(!in)
-  {
-    if (!inifile_unavailable)
-    {
-      char msg[80];
-      inifile_unavailable = 1;
-      sprintf(msg, "Cannot load INI file \"%.55s\"", inifile);
-      perror(msg);
+    FILE* in;
+    char buf[100];
```

Comment:
```suggestion
    char buf[101];
```
We should leave 1 byte for the null terminator (or call `fgets(buf, 99, in)` below instead). It might be a good idea to fix this in the same PR since this part of the code is already being updated.